znoyder/get_project_zuul_files.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Red Hat, Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+import json
+import requests
+
+from argparse import ArgumentParser
+from functools import partial
+from multiprocessing import Pool, cpu_count
+from pathlib import Path
+from sys import exit
+
+
+URL_BASE_API = "https://opendev.org/api/v1/repos/openstack"
+LIST_SPECIFIC_FOLDER_CONTENT_ENDPOINT = URL_BASE_API + "/{project_name}/contents/{folder}?ref={branch}"
+
+
+def process_arguments() -> tuple:
+    """ Return the branch, destination folder and project name of the processed arguments"""
+    parser = ArgumentParser()
+    parser.add_argument(
+        '-p', '--projectname',
+        dest='project_name',
+        help="Insert the project name",
+        metavar="PROJECT_NAME",
+        required=True
+    )
+    parser.add_argument(
+        '-b', '--branchname',
+        dest='branch_name',
+        help="Select branch to work",
+        metavar="BRANCH_NAME",
+        required=True
+    )
+    parser.add_argument(
+        '-d', '--destinationfolder',
+        dest='destination_folder',
+        help="Insert the destination folder",
+        metavar="DESTINATION_FOLDER",
+        required=True
+    )
+    arguments = parser.parse_args()
+    return arguments
+
+
+def __createFolder(folder_name: str) -> None:
+    Path(folder_name).mkdir(parents=True, exist_ok=True)
```

Comment:
This function is just a single and it is used only once, so I think you an just move the code to where it is called...